src/controller/controller.lua

```diff
@@ -0,0 +1,58 @@
+local Timer = require("timer/timer")
+local utils = require("utils/utils")
+local settings = require("settings/settings")
+
+local Controller = {}
+
+function Controller:new(board)
+    local controller = {
+        board = board,
+        timers = {
+            ['moveLeft'] = Timer:new(),
+            ['moveRight'] = Timer:new(),
+            ['moveDown'] = Timer:new(),
+        },
+    }
+    self.__index = self
+    setmetatable(controller, self)
+    return controller
+end
+
+function Controller:run(action)
+    if action == 'moveLeft' then self.board.movement:moveLeft() end
+    if action == 'moveRight' then self.board.movement:moveRight() end
+    if action == 'moveDown' then self.board.movement:moveDown() end
+    if action == 'rotateCw' then self.board.movement:rotateCw() end
+    if action == 'rotateCcw' then self.board.movement:rotateCcw() end
+    if action == 'hardDrop' then self.board.movement:hardDrop() end
+    if action == 'hold' then self.board:holdCurrentTetromino() end
+end
+
+function Controller:handleNonRepeatKeys(key, scancode, isrepeat)
```

Comment:
```suggestion
function Controller:handleNonRepeatKeys(key, scancode, isrepeat)
    self:run(key)
end
```

Reply:
Don't need the for loop after the changes.

Reply:
I'll probably need another function in between to get from the key to the action. 
`self:run(self:getAction(key))`. But then the getAction function just becomes the same as what I had in this function except it returns the action instead of running it.

Reply:
Okay. If you want to be able to change keys can you just swap the key and value in your settings? So instead of moveleft= left just do left = moveleft. That way it's still a one liner.

Reply:
Can you just flip the keys and values in the settings file and then just call self:run(settings.keybindings[key])???